src/BatteryStateOfChargeService.h

```diff
@@ -42,8 +42,13 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
     void addData(const BatteryData& batteryData);
 
 private:
-    // Convention is to have an _ after private members
     const double initialStateOfChargePercent_;
-
-    // Recommend adding necessary variables here
+    double newCurrent_;
+    double totalCurrent_;
+    QTime estimatedTime_;
+    double oldCurrent_;
+    QTime oldTime_ = QTime(0,0,0);
+    QTime newTime_;
+    double totalAmpUsed_ = 0;
+    double initialAmpUsed_;
```

Comment:
You have a couple private members here that are not being used anymore. 

You also are trying to initialize variables here, this is not valid C++ in some compilers and is also not good practice. It is better to initialize them in the constructor.